src/validation/userValidation.js

```diff
@@ -0,0 +1,96 @@
+import joi from '@hapi/joi';
+import passwordComplexity from 'joi-password-complexity';
+import { response, ApiResponse } from '../utils';
+import UserService from '../services/index';
+
+// password complexity object
+const complexityOPtions = {
+  min: 8,
+  max: 250,
+  lowerCase: 1,
+  upperCase: 1,
+  numeric: 1,
+  symbol: 1,
+  requirementCount: 3,
+};
+
+// TODO: create userValidation class
+/**
+   * This class holds all methods used for user validation
+   * Functions:
+   * 1) signup - validates user upon registration.
+   * 2) dummy - a dummy method for testing validations upon success.
+   */
+export default class userValidation {
+  /**
+     * Validates user paramenters upon registration
+     *
+     * @param {object} req - The request from the endpoint.
+     * @param {object} res - The response returned by the method.
+     * @param {object} next - the returned values going into the next operation.
+     * @returns {object} - returns an object (error or response).
+     */
+  static async signup(req, res, next) {
```

Comment:
Validations in the validation folder should not be used as a middleware. Validation methods should be created here and imported in a middleware file.